app/scripts/lib/modal_manager.js

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'jquery',
+  'velocity',
+  'velocityui'
+], function (_, $) {
+  'use strict';
+
+  function ModalManager () {
+    this._viewStates = [];
+    this.$modal = $('#modal');
+
+    // Listen for all clicks
+    $(document).on('click', _.bind(this._closeOnClick, this));
+
+    // Close modal on resize to prevent weirdness between desktop and mobile
+    $(document).on('resize', _.debounce(_.bind(this.close, this), 100));
+
+    // Handle default button actions
+    this.$modal.on('click', 'a.close', _.bind(this.close, this));
+    this.$modal.on('click', 'a.back', _.bind(this.pop, this));
+  }
+
+  _.extend(ModalManager.prototype, {
+    open: function (view, options) {
+      this._destroyViews();
+
+      this.push(view, options);
+    },
+
+    push: function (view, options) {
+      this._viewStates.push({ view: view, options: (options || {}) });
+
+      this._show();
+    },
+
+    pop: function () {
+      var viewState = this._viewStates.pop();
+
+      if (viewState) {
+        viewState.view.destroy();
+      }
+
+      if (this._viewStates.length > 0) {
+        this._show();
+      } else {
+        this._hide();
+      }
+    },
+
+    close: function (options) {
+      if (!options) {
+        options = {};
+      }
+
+      this._destroyViews();
+      this._hide(options);
+    },
+
+    _destroyViews: function () {
+      _.each(this.viewStates, function (viewState) {
+        viewState.view.destroy();
+      });
+
+      this._viewStates = [];
+    },
+
+    _show: function () {
+      var viewState = _.last(this._viewStates);
+
+      viewState.view.render();
+
+      // Force delegate events to fix an issue where restoring a previous view breaks event bindings
+      viewState.view.delegateEvents();
+
+      // TODO: Add fancy positioning: likely in the middle of the screen. We're only supporting
+      // full screen right now, so it can come later.
+
+      // Add full-screen class if the full screen option is enabled
+      if (viewState.options.fullScreen) {
+        this.$modal.addClass('full-screen');
+      } else {
+        this.$modal.removeClass('full-screen');
+      }
+
+      // Replace modal contents
+      this.$modal.html(viewState.view.el);
+
+      // Animate if an effect is provided
+      if (viewState.options.effect) {
+        this.$modal.velocity(viewState.options.effect);
+      } else {
+        this.$modal.show();
+      }
+    },
+
+    _hide: function (options) {
+      // Animate if an effect is provided
+      if (options.effect) {
```

Comment:
Unlikely since you wouldn't want something like a `fadeIn` effect when closing the modal. We'll probably have some default effects for the modals which would become the fallback.